src/server/common_services/auth_server.js

```diff
@@ -681,13 +681,18 @@ function has_bucket_anonymous_permission(bucket) {
         Effect: "Allow",
         Principal: "*",
         Action: ["s3:GetObject"],
-        Resource: [`arn:aws:s3:::${bucket.name}/*`],
+        Resource: [`arn:aws:s3:::${bucket.name.unwrap()}/*`],
     };
-    const bucket_policy = bucket.policy;
+    const bucket_policy = bucket.s3_policy;
     if (!bucket_policy) return false;
-    const statements = bucket_policy.Statement;
+    const statements = bucket_policy.statement;
     if (!statements) return false;
-    return _.some(statements, record => _.isEqual(required_configuration, record));
+    return _.some(statements, record => {
```

Comment:
can you please add a comment that explains what we are checking here: one of the priniciples is * and the rest looks like required_configuration, have to say it can be done much simpler without going to JSON and return back. maybe I'm missing something... what about if action has more than just s3:getObject, or have other resources (which is unneeded I agree - but maybe was added anyway)?

Reply:
Refactored the code.